invenio_rdm_records/services/services.py

```diff
@@ -289,6 +301,13 @@ def restore_record(self, identity, id_, expand=False, uow=None):
         # Commit and reindex record
         uow.register(RecordCommitOp(record, indexer=self.indexer))
 
+        # reindex associated draft
+        try:
+            draft = self.draft_cls.pid.resolve(id_)
+            uow.register(RecordIndexOp(draft, indexer=self.indexer))
```

Comment:
```suggestion
            uow.register(RecordIndexOp(draft, indexer=self.draft_indexer))
```

I recently found out, this is a thing :)

ditto above I guess when deleting from the index.